src/3d/qgs3dmapscene.cpp

```diff
@@ -364,15 +364,19 @@ void addQLayerComponentsToHierarchy( Qt3DCore::QEntity *entity, const QVector<Qt
   }
 }
 
-void Qgs3DMapScene::updateScene()
+void Qgs3DMapScene::updateScene( bool forceUpdate )
 {
-  QgsEventTracing::addEvent( QgsEventTracing::Instant, QStringLiteral( "3D" ), QStringLiteral( "Update Scene" ) );
+  if ( forceUpdate )
+    QgsEventTracing::addEvent( QgsEventTracing::Instant, QStringLiteral( "3D" ), QStringLiteral( "Update Scene" ) );
 
   for ( Qgs3DMapSceneEntity *entity : std::as_const( mSceneEntities ) )
   {
-    entity->handleSceneUpdate( buildSceneContext() );
-    if ( entity->hasReachedGpuMemoryLimit() )
-      emit gpuMemoryLimitReached();
+    if ( forceUpdate || ( entity->isEnabled() && entity->needsUpdate() ) )
+    {
+      entity->handleSceneUpdate( buildSceneContext() );
```

Comment:
Should we move buildSceneContext() out of the loop so that we don't have to call this for every entity?

Reply:
Good idea. Done.